### PR TITLE
feat: containerize performance tests [INFENG-222]

### PR DESCRIPTION
**Pull request by loksonarius**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: .github/workflows/build-perf-image-perf-test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: Create and publish a Docker image for performance tests
+on:
+  pull_request:
+    paths:
+      - 'performance/**'
+  push:
+    branches:
+      - 'main'
```
Comment (dannysauer) [inline .github/workflows/build-perf-image-perf-test.yml:9]: Should this have a paths filter as well?  We probably don't need to rebuild on every push to main. :)

File: performance/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Copy over all source code
+COPY . .
+
+# Install build dependencies
+RUN npm install
+
+# Run build scripts for test scripts
+RUN npm start
+
+FROM grafana/k6:${K6_VERSION} as base
```
Comment (dannysauer) [inline performance/Dockerfile:18]: Doing it this way means Dependabot can't automatically open update PRs for us, and we won't get notified of security issues. Do we anticipate needing to run this with different container versions often enough that editing the Dockerfile is more painful than manually keeping track of both upstream base images?

Comment (loksonarius) [reply performance/Dockerfile:18]: I figure letting dependabot bump automatically is gonna yield better results than making the version easier for humans to bump. I'll switch em over.

File: .github/workflows/build-perf-image-perf-test.yml (hunk 1 above)
Comment (loksonarius) [reply .github/workflows/build-perf-image-perf-test.yml:9]: Yeah, it really should.

**Review by dannysauer — APPROVED** [review]
Thanks.  Of course, we'll have to update the Dependabot config to look in this directory for a Dockerfile now, too. ;)

**Pull request merged**